modules/renter/contractor/update.go

```diff
@@ -90,12 +90,17 @@ func (c *Contractor) ProcessConsensusChange(cc modules.ConsensusChange) {
 
 	c.mu.Lock()
 	for _, block := range cc.RevertedBlocks {
-		if block.ID() != types.GenesisID {
-			c.blockHeight--
-		}
 		// Remove recoverable contracts found in reverted block.
 		c.removeRecoverableContracts(block)
 	}
+
+	// Calculate the height before the blocks were applied. If this is the
+	// genesis change set the height to 0.
+	c.blockHeight = cc.BlockHeight - types.BlockHeight(len(cc.AppliedBlocks))
+	if cc.BlockHeight == 0 {
+		c.blockHeight = 0
+	}
```

Comment:
Maybe add a `ConsensusChange.InitialHeight()` method to simplify this? Or `PreChangeHeight` and `PostChangeHeight` fields? 🤔